File: go/src/leetcode/offer/Offer55_1.go
package offer

/**
 * @author wujingxinit@outlook.com
 * @date 2022/12/30 23:54
 */

func maxDepth(root *TreeNode) int {

	if root == nil {
		return 0
	}
	return Max(maxDepth(root.Left), maxDepth(root.Right)) + 1
}

File: java/src/leetcode/lists/offer/Offer55_1.java
package leetcode.lists.offer;

import common.TreeNode;

/**
 * @author jingxinwu
 * @date 2021-11-25 12:33 上午
 */
public class Offer55_1 {

    public int maxDepth(TreeNode root) {
        if (root == null) {
            return 0;
        }
        return Math.max(maxDepth(root.left), maxDepth(root.right)) + 1;
    }
}
